compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -4111,32 +4139,45 @@ static const TR::ILOpCodes MemoryLoadOpCodes[TR::NumOMRTypes] =
 
 TR::InstOpCode OMR::X86::TreeEvaluator::getNativeSIMDOpcode(TR::ILOpCodes opcode, TR::DataType type, bool memForm)
    {
-   BinaryArithmeticOps arithmetic;
+   ArithmeticOps binaryOp = ArithmeticInvalid;
+   ArithmeticOps unaryOp = ArithmeticInvalid;
 
    if (OMR::ILOpCode::isVectorOpCode(opcode))
       {
       switch (OMR::ILOpCode::getVectorOperation(opcode))
          {
          case TR::vadd:
-            arithmetic = BinaryArithmeticAdd;
+            binaryOp = BinaryArithmeticAdd;
             break;
          case TR::vsub:
-            arithmetic = BinaryArithmeticSub;
+            binaryOp = BinaryArithmeticSub;
             break;
          case TR::vmul:
-            arithmetic = BinaryArithmeticMul;
+            binaryOp = BinaryArithmeticMul;
             break;
          case TR::vdiv:
-            arithmetic = BinaryArithmeticDiv;
+            binaryOp = BinaryArithmeticDiv;
             break;
          case TR::vand:
-            arithmetic = BinaryArithmeticAnd;
+            binaryOp = BinaryArithmeticAnd;
             break;
          case TR::vor:
-            arithmetic = BinaryArithmeticOr;
+            binaryOp = BinaryArithmeticOr;
             break;
          case TR::vxor:
-            arithmetic = BinaryArithmeticXor;
+            binaryOp = BinaryArithmeticXor;
+            break;
+         case TR::vmin:
+            binaryOp = BinaryArithmeticMin;
```

Comment:
vmin and vmax opcodes violate java's floating-point semantics. For instance, if one operand is NaN, the result must be NaN.